src/App.tsx

```diff
@@ -158,7 +180,7 @@ class Square extends React.Component<SquareProps, {}> {
     render() {
         let light : string = this.props.isHighlighted ? "highlight" : (this.props.isContradicting ? "contradict" : "");
         let marks : boolean[] = this.getMarks();
-        if (this.props.markings && this.props.markings.length &&!this.props.isPermanent) {
+        if (this.props.markings && this.props.markings.length &&!this.props.isPermanent && !this.props.value) {
```

Comment:
1. Fix whitespace
2. This logic is too confusing to me. Please either add a comment or add a helper function that makes it more clear why you need so many conditions in this if